frontend/src/metabase/dashboard/actions.js

```diff
@@ -202,6 +202,15 @@ export const setMultipleDashCardAttributes = createAction(
   SET_MULTIPLE_DASHCARD_ATTRIBUTES,
 );
 
+function generateTemporaryDashcardId() {
+  return Math.random();
+}
+
+// real dashcard ids are integers >= 1
+function isNewDashcard(dashcard) {
```

Comment:
Could we remove id generation and check for missing `dashboard.id` here?

Reply:
New/unsaved dashcards are stored alongside real/saved dashcards in redux, so they need some sort of id to be identifiable. Using `undefined` as an `id` means you can't add multiple new cards to a dashboard at once, so we'd need to change how things are stored before getting rid of the generated ids. Getting rid of this logic seems reasonable to me, but let's not do it in this PR, if you don't mind.

Reply:
Makes sense.